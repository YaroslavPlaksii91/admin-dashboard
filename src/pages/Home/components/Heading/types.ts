export type HeadingProps = {
  sortKey: string;
  sortDirection: 'asc' | 'desc';
  handleSort: (name: string) => void;
  columns: readonly {
    title: string;
    name: string;
    size: number;
  }[];
};
