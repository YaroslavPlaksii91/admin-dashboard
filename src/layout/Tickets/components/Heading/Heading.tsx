import { FC } from 'react';
import { Grid } from '@mui/material';

export const Heading: FC = () => {
  return (
    <Grid
      container
      sx={{
        padding: '0 32px 12px',
        fontSize: 14,
        fontWeight: 700,
        lineHeight: 1.3,
        letterSpacing: 0.2,
        color: 'secondaryTextColor',
        borderBottom: '2px solid',
        borderBottomColor: 'borderColor',
      }}
    >
      <Grid item xs={4}>
        Ticket details
      </Grid>
      <Grid item xs={3}>
        Customer name
      </Grid>
      <Grid item xs={3}>
        Date
      </Grid>
      <Grid item xs={2}>
        Priority
      </Grid>
    </Grid>
  );
};
