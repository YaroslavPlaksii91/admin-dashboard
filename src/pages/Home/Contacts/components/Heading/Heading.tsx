import { FC } from 'react';
import { Grid } from '@mui/material';

export const Heading: FC = () => {
  return (
    <Grid
      container
      sx={{
        padding: '0 32px 12px',
        fontSize: 14,
        fontWeight: 700,
        lineHeight: 1.3,
        letterSpacing: 0.2,
        color: 'secondaryTextColor',
        borderBottom: '2px solid',
        borderBottomColor: 'borderColor',
      }}
    >
      <Grid item xs={4}>
        Name
      </Grid>
      <Grid item xs={3}>
        Email
      </Grid>
      <Grid item xs={3}>
        Address
      </Grid>
      <Grid item xs={2}>
        Created at
      </Grid>
    </Grid>
  );
};
