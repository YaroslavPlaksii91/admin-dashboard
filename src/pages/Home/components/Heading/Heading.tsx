import { FC } from 'react';
import { Grid } from '@mui/material';

import { ColumnHeading } from '@pages/Home/components/ColumnHeading/ColumnHeading';

import { HeadingProps } from './types';

export const Heading: FC<HeadingProps> = ({
  handleSort,
  sortKey,
  sortDirection,
  columns,
}) => {
  return (
    <Grid
      container
      sx={{
        padding: '0 32px 12px',
        fontSize: 14,
        fontWeight: 700,
        lineHeight: 1.3,
        letterSpacing: 0.2,
        color: 'secondaryTextColor',
        borderBottom: '2px solid',
        borderBottomColor: 'borderColor',
        cursor: 'pointer',
      }}
    >
      {columns.map(col => (
        <ColumnHeading
          key={col.name}
          size={col.size}
          onClick={() => handleSort(col.name)}
          name={col.name}
          title={col.title}
          sortKey={sortKey}
          sortDirection={sortDirection}
        />
      ))}
    </Grid>
  );
};
