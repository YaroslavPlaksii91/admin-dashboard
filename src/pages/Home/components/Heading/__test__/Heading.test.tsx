import { cleanup, render, screen, fireEvent } from '@testing-library/react';

import { Heading } from '../Heading';

describe('Heading component', () => {
  afterEach(cleanup);

  const handleSort = jest.fn();
  const columns = [
    { name: 'column1', title: 'Column 1', size: 4 },
    { name: 'column2', title: 'Column 2', size: 4 },
    { name: 'column3', title: 'Column 3', size: 4 },
  ];

  it('should be rendered', () => {
    const { asFragment } = render(
      <Heading
        handleSort={handleSort}
        sortKey="column1"
        sortDirection="asc"
        columns={columns}
      />,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders correctly with column headings', () => {
    render(
      <Heading
        handleSort={handleSort}
        sortKey="column1"
        sortDirection="asc"
        columns={columns}
      />,
    );

    columns.forEach(col => {
      expect(screen.getByText(col.title)).toBeInTheDocument();
    });
  });

  it('calls handleSort when a column heading is clicked', () => {
    render(
      <Heading
        handleSort={handleSort}
        sortKey="column1"
        sortDirection="asc"
        columns={columns}
      />,
    );

    columns.forEach(col => {
      fireEvent.click(screen.getByText(col.title));
      expect(handleSort).toHaveBeenCalledWith(col.name);
    });
  });
});
