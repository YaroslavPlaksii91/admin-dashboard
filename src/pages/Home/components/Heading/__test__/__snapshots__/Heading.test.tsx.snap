// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading component should be rendered 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-container css-bxay1f-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
    >
      Column 1 
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-1j6x8sl-MuiSvgIcon-root"
        data-testid="NorthIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="m5 9 1.41 1.41L11 5.83V22h2V5.83l4.59 4.59L19 9l-7-7-7 7z"
        />
      </svg>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
    >
      Column 2 
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
    >
      Column 3 
    </div>
  </div>
</DocumentFragment>
`;
